app/javascript/controllers/sep_types_list_controller.js

```diff
@@ -0,0 +1,187 @@
+    let originalPosition = parseInt(card.dataset.index);
+    
+    let cards = $(this.qleListTarget).find('.card').get();
+    if (cards[originalPosition].dataset.id === card.dataset.id) { // If the card was not moved, do not sort the list
+      return;
+    }
+
+    // map the card elements to an array of card ids and their ordinal positions
+    let enumeratedCards = cards.entries();
+    for (const [index, card] of enumeratedCards) {
+      card.dataset.index = index;
+      card.dataset.ordinal_position = index + 1;
+    }
+    let sortData = [...enumeratedCards].map((entry) => { return { id: entry[1].dataset.id, position: entry[0] + 1 } });
+    super.updateList({sort_data: sortData})
+      .then(data => {
+        let isSuccess = data['status'] === "success";
+        if (bs4) {
+          var flashDiv = super.showBanner(isSuccess);
+        } else {
+          var flashDiv = $("#sort_notification_msg");
+          var flashHeader = flashDiv.find(".toast-header");
+          flashDiv.show();
+          flashDiv.toggleClass("success", isSuccess);
+          flashDiv.removeClass("error", !isSuccess);
+          flashHeader.toggleClass("success", isSuccess);
+          flashHeader.removeClass("error", !isSuccess);
+          
+          flashHeader.find("strong").text(data['message']);
+          flashDiv.find(".toast-body").text(event.item.textContent);
+        }
+
+        setTimeout(function() {
+          flashDiv.addClass('hidden');
+        }, 3500);
+		})
+	}
+}
+
+// Manages the commonality threshold input and marker for the SEP types list.
+class UpdateThresholdManager extends UpdateListManager {
+
+  constructor(marketKind, qleListTarget, thresholdInputTarget) {
+    super(marketKind);
+    this.thresholdInputTarget = thresholdInputTarget;
+    this.qleList = $(qleListTarget);
+  }
+
+  /** 
+  * Handle updating the commonality threshold for a QLE list.
+  * Performs the PATCH request, then updates the threshold marker element and shows the banner based on the response.
+  */
+  set() {
+    let threshold = this.thresholdInputTarget.value;
+
+    super.updateList({commonality_threshold: threshold})
+      .then(data => {
+        let isSuccess = data['status'] === 'success';
+        if (isSuccess) {
+          this.thresholdInputTarget.dataset.initialValue = threshold;
+          super.showBanner(true);
+
+          // move the threshold marker to the new threshold index or hide it if out of bounds
+          let thresholdMarker = this.qleList.find('#threshold-marker').removeClass('hidden');
+          let newBoundaryQLE = this.qleList.find(`.card:eq(${threshold})`);
+          if (newBoundaryQLE.length) {
+            thresholdMarker.detach().insertBefore(newBoundaryQLE);
+          } else {
+            thresholdMarker.addClass('hidden');
+          }
+        } else {
+          this.thresholdInputTarget.value = this.thresholdInputTarget.dataset.initialValue;
+          super.showBanner(false);
+        }
+      })
+  }
+}
```

Comment:
Would it be possible to have each of these non-controller JS classes separated into their own respective files? Maybe under `app/javascript/src/sep_types_list_managers/` or something along those lines?

Reply:
It is possible, but I actually think this is fine. Think of this like a rails controller with 2 separate actions.